addons/l10n_sa_edi/models/account_edi_xml_ubl_21_zatca.py

```diff
@@ -132,15 +132,18 @@ def _get_partner_address_vals(self, partner):
             'plot_identification': partner.l10n_sa_edi_plot_identification,
         }
 
-    def _export_invoice_filename(self, invoice):
+    def _export_invoice_filename(self, invoice, file_format='xml'):
         """
             Generate the name of the invoice XML file according to ZATCA business rules:
             Seller Vat Number (BT-31), Date (BT-2), Time (KSA-25), Invoice Number (BT-1)
         """
         vat = invoice.company_id.partner_id.commercial_partner_id.vat
-        invoice_number = re.sub("[^a-zA-Z0-9 -]", "-", invoice.name)
+        invoice_number = re.sub(r'[^a-zA-Z0-9 -]+', '-', invoice.name)
         invoice_date = fields.Datetime.context_timestamp(self.with_context(tz='Asia/Riyadh'), invoice.l10n_sa_confirmation_datetime)
-        return '%s_%s_%s.xml' % (vat, invoice_date.strftime('%Y%m%dT%H%M%S'), invoice_number)
+        file_name = '%s_%s_%s' % (vat, invoice_date.strftime('%Y%m%dT%H%M%S'), invoice_number)
+        if file_format:
+            file_name += '.%s' % file_format
```

Comment:
f string for the win!
```suggestion
        file_name = f"{vat}_{invoice_date.strftime('%Y%m%dT%H%M%S')}_{invoice_number}"
        if file_format:
            file_name = f'{file_name}.{file_format}'
```